scipy/cluster/hierarchy.py

```diff
@@ -4082,8 +4139,9 @@ def leaders(Z, T):
     array([1, 2, 3, 4], dtype=int32)
 
     """
-    Z = np.asarray(Z, order='C', dtype=np.float64)
-    T = np.asarray(T, order='C')
+    xp = array_namespace(Z, T)
+    Z = as_xparray(Z, order='C', dtype=xp.float64)
+    T = as_xparray(T, order='C')
```

Comment:
one issue snuck in here, the `if type(T) != np.ndarray ` below is causing a test failure